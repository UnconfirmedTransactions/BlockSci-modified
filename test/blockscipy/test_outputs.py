import pytest


def test_p2pkh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2pkh-spend-2-tx"])
    output_regression(tx, regtest)


def test_p2sh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2sh-spend-0-tx"])
    output_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wpkh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wpkh-spend-1-tx"])
    output_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wsh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wsh-spend-1-tx"])
    output_regression(tx, regtest)


def output_regression(tx, regtest):
    for out in tx.outputs:
        print(out.address, file=regtest)
        print(out.address_type, file=regtest)
        print(out.block, file=regtest)
        print(out.index, file=regtest)
        print(out.is_spent, file=regtest)
        print(out.spending_input, file=regtest)
        print(out.spending_tx, file=regtest)
        print(out.spending_tx_index, file=regtest)
        print(out.tx, file=regtest)
        print(out.tx_index, file=regtest)
        print(out.value, file=regtest)

def test_output_input_mapping(chain):
    for block in chain:
        for tx in block:
            for output in tx.outputs:
                if output.is_spent:
                    assert output == output.spending_input.spent_output
                    assert tx == output.spending_input.spent_tx
