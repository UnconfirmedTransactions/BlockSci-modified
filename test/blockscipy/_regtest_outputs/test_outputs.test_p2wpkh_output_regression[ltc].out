WitnessPubkeyAddress(rltc1qxxhhtzpwdfnvckc5w6p89ulje744wfdjnptlx9)
Pay to witness pubkey hash
Block(tx_count=3, height=118, header_hash=c791fd0723aae127e0eaea4faae859388cbf2f83316e64ff8be520e06d5ca673, version=536870912, timestamp=1535765400, bits=545259519, nonce=0)
0
True
TxIn(spent_tx_index=134, address=WitnessPubkeyAddress(rltc1qxxhhtzpwdfnvckc5w6p89ulje744wfdjnptlx9), value=100000000)
Tx(len(txins)=1, len(txouts)=1, size_bytes=113, block_height=118, tx_index=135)
135
Tx(len(txins)=1, len(txouts)=2, size_bytes=222, block_height=118, tx_index=134)
134
100000000
PubkeyHashAddress(mjTkW3DjgyZck4KbiRusZsqTgaYTxdSz6z)
Pay to pubkey hash
Block(tx_count=3, height=118, header_hash=c791fd0723aae127e0eaea4faae859388cbf2f83316e64ff8be520e06d5ca673, version=536870912, timestamp=1535765400, bits=545259519, nonce=0)
1
False
None
None
None
Tx(len(txins)=1, len(txouts)=2, size_bytes=222, block_height=118, tx_index=134)
134
4899990000
